application/src/main/webapp/js/chat.js

```diff
@@ -120,6 +120,17 @@ var chatApplication = new ChatApplication();
     });
 
     $('#msg').keydown(function(event) {
+      //prevent the default behavior of the enter button
+      if ( event.which == 13 ) {
+        event.preventDefault();
+      }
+      //adding (shift or ctl or alt) + enter for adding carriage return in a specific cursor
+      if ( event.keyCode == 13 && (event.shiftKey||event.ctrlKey||event.altKey) ) {
+        this.value = this.value.substring(0, this.selectionStart)+"\n"+this.value.substring(this.selectionEnd,this.value.length);
+        $("#msg").animate({
+          scrollTop:$("#msg")[0].scrollHeight - $("#msg").height()
+        },350,function(){/* Animation complete.*/});
```

Comment:
The new line should appear instantaneously, so please remove this 350ms delay (do we even need to use the animate function in this case ?), it must be reactive. And you can also remove the empty function since it does nothing and pollutes.